.github/workflows/Github_pytest.yml

```diff
@@ -95,10 +89,94 @@ jobs:
       - name: Parallel tests with pytest
         uses: ./.github/actions/pytest_parallel
 
+  Cuda:
+
+    runs-on: ubuntu-latest
+    container: nvidia/cuda:11.7.1-devel-ubuntu20.04
+    if: github.event.pull_request.base.ref != 'master'
+    steps:
+      - uses: actions/checkout@v2
+      - name: Prepare docker
+        run: |
+          apt update && apt install sudo
+          TZ=Europe/France
+          ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
+          DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends tzdata
+        shell: bash
+      - name: CUDA Version
+        run: nvcc --version # cuda install check
+      - name: Install dependencies
+        uses: ./.github/actions/linux_install
+      - name: Install python dependencies
+        uses: ./.github/actions/pip_installation
+      - name: Coverage install
+        uses: ./.github/actions/coverage_install
+      - name: Ccuda tests with pytest
+        uses: ./.github/actions/pytest_run_cuda
+      - name: Collect coverage information
+        continue-on-error: True
+        uses: ./.github/actions/coverage_collection
+      - name: Save code coverage report
+        uses: actions/upload-artifact@v3
+        with:
+          name: cuda-coverage-artifact
+          path: .coverage
+          retention-days: 1
+
+  CoverageCollection:
+
+    runs-on: ubuntu-latest
+    needs: [Linux, Cuda]
+    if: ${{ always() && needs.Linux.result == 'success' && needs.Cuda.result != 'failure' }}
```

Comment:
is there more that `success`, `failure` in the result ?

Reply:
It can also be `skipped`